flink-core/src/main/java/org/apache/flink/util/FileUtils.java

```diff
@@ -56,6 +60,15 @@
 	/** The length of the random part of the filename. */
 	private static final int RANDOM_FILE_NAME_LENGTH = 12;
 
+	/**
+	 * The maximum size of array to allocate for reading. See
+	 * {@link java.nio.file.Files#MAX_BUFFER_SIZE} for more.
+	 */
+	private static final int MAX_BUFFER_SIZE = Integer.MAX_VALUE - 8;
+
+	/** The size of the buffer used for reading. */
+	private static final int BUFFER_SIZE = 4096;
```

Comment:
Math.min(4096, new Long(sun.misc.VM.maxDirectMemory()).intValue()); ?

Reply:
The sun.misc package is Sun's dedicated api, and it seems that it should be avoided to use as much as possible. Currently in most cases, the size of direct memory should be larger than 4K.

Reply:
Yes, Oracle suggests that developers should not write programs that call 'sun' packages.  Adding the comparison of `maxDirectMemory` is an enhancement of program robustness, but I agree that the value of `maxDirectMemory` will usually be greater than 4K.